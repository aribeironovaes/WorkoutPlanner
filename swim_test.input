(100yards, 100)(10s, 0)
-- EOT
(10min, *, TT)
-- EOT
(500y, 100, free)
-- EOT
2[(200yards, 100, neg split, 30s)]
-- EOT
(500y, *, free)
-- EOT
(500yards, 90, 50sec, warmup)
-- EOT
(400yards, +10s, 30sec, warmup)
-- EOT
"Comment 123"
-- EOT
