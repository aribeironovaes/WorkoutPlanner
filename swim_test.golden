Input: 
(100yards, 100)(10s, 0)
Normalized: 
(100yards, 100)
(10sec, 0)
AST: 
ArrayInterval(
	SimpleInterval(100yards, 100%, )
	SimpleInterval(10sec, 0%, )
)
IF (Avg): 0.9459053029269173
TSS: 2.4
TSS2: 2.5
Dominant Unit: %
Pretty Print:
	* 100 on 1'25''
	* 10'' rest
Duration (Sec): 95
----------------------------
Input: 
(10min, *, TT)
Normalized: 
(10min, *, TT)
AST: 
ArrayInterval(
	SimpleInterval(10min, *, TT)
)
IF (Avg): 0.88
TSS: 12.9
TSS2: 13.2
Dominant Unit: free-ride
Pretty Print:
	* 10' TT
Duration Unit: min
Duration (Sec): 600
----------------------------
Input: 
(500y, 100, free)
Normalized: 
(500yards, 100, free)
AST: 
ArrayInterval(
	SimpleInterval(500yards, 100%, free)
)
IF (Avg): 1
TSS: 11.8
TSS2: 11.8
Dominant Unit: %
Pretty Print:
	* 500 free on 7'05''
Duration Unit: yards
Yards: 500
Duration (Sec): 425
----------------------------
Input: 
2[(200yards, 100, neg split, 30s)]
Normalized: 
2[(200yards, 100, neg split, 30sec)]
AST: 
ArrayInterval(
	RepeatInterval(count=2, 
		SimpleInterval(200yards, 100%, neg split, 30sec)
	)
)
IF (Avg): 1
TSS: 9.4
TSS2: 10.3
Dominant Unit: %
Pretty Print:
	* 2 x 200 neg split on 2'50'' off 3'20''
Duration Unit: yards
Yards: 400
Duration (Sec): 370
----------------------------
Input: 
(500y, *, free)
Normalized: 
(500yards, *, free)
AST: 
ArrayInterval(
	SimpleInterval(500yards, *, free)
)
IF (Avg): 0.88
TSS: 10.4
TSS2: 10.6
Dominant Unit: free-ride
Pretty Print:
	* 500 free
Duration Unit: yards
Yards: 500
Duration (Sec): 482.9545454545454
----------------------------
Input: 
(500yards, 90, 50sec, warmup)
Normalized: 
(500yards, 90, warmup, 50sec)
AST: 
ArrayInterval(
	SimpleInterval(500yards, 90%, warmup, 50sec)
)
IF (Avg): 0.9000000000000001
TSS: 10.6
TSS2: 11.8
Dominant Unit: %
Pretty Print:
	* 500 warmup on 7'52'' off 8'42''
Duration Unit: yards
Yards: 500
Duration (Sec): 522.2222222222222
----------------------------
Input: 
(400yards, +10s, 30sec, warmup)
Normalized: 
(400yards, +10s, warmup, 30sec)
AST: 
ArrayInterval(
	SimpleInterval(400yards, CSS+10, warmup, 30sec)
)
IF (Avg): 0.8947368421052629
TSS: 8.5
TSS2: 9.2
Dominant Unit: offset-seconds
Pretty Print:
	* 400 warmup on 6'20'' off 6'50'' (+10)
Duration Unit: yards
Yards: 400
Duration (Sec): 410.0000000000001
----------------------------
Input: 
"Comment 123"
Normalized: 
"Comment 123"
AST: 
ArrayInterval(
	Comment(Comment 123)
)
IF (Avg): 0
TSS: 0
TSS2: 0
Duration (Sec): 0
----------------------------
Input: 
(10min, 100, 2min, CSS)
Normalized: 
(10min, 100, CSS, 2min)
AST: 
ArrayInterval(
	SimpleInterval(10min, 100%, CSS, 2min)
)
IF (Avg): 1
TSS: 16.7
TSS2: 20
Dominant Unit: %
Pretty Print:
	* 10' CSS on 10' off 12'
Duration Unit: min
Duration (Sec): 720
----------------------------
Input: 
(500m, 100, warmup)
Normalized: 
(500m, 100, warmup)
AST: 
ArrayInterval(
	SimpleInterval(500m, 100%, warmup)
)
IF (Avg): 1
TSS: 12.9
TSS2: 12.9
Dominant Unit: %
Pretty Print:
	* 500 warmup on 7'44''
Duration Unit: m
Meters: 500
Duration (Sec): 464.7856517935258
----------------------------
Input: 
(500m, warmup, 100)
Normalized: 
(500m, 100, warmup)
AST: 
ArrayInterval(
	SimpleInterval(500m, 100%, warmup)
)
IF (Avg): 1
TSS: 12.9
TSS2: 12.9
Dominant Unit: %
Pretty Print:
	* 500 warmup on 7'44''
Duration Unit: m
Meters: 500
Duration (Sec): 464.7856517935258
----------------------------
Input: 
(500m, 100, warmup), (200m, 80, easy)
Normalized: 
(500m, 100, warmup)
(200m, 80, easy)
AST: 
ArrayInterval(
	SimpleInterval(500m, 100%, warmup)
	SimpleInterval(200m, 80%, easy)
)
IF (Avg): 0.938083151964686
TSS: 17
TSS2: 16.4
Dominant Unit: %
Pretty Print:
	* 500 warmup on 7'44''
	* 200 easy @ 80%
Duration Unit: m
Meters: 700
Duration (Sec): 697.1784776902887
----------------------------
Input: 
(500yards, 100, warmup)
Normalized: 
(500yards, 100, warmup)
AST: 
ArrayInterval(
	SimpleInterval(500yards, 100%, warmup)
)
IF (Avg): 1
TSS: 11.8
TSS2: 11.8
Dominant Unit: %
Pretty Print:
	* 500 warmup on 7'05''
Duration Unit: yards
Yards: 500
Duration (Sec): 425
----------------------------
Input: 
(500yards, warmup, 100)
Normalized: 
(500yards, 100, warmup)
AST: 
ArrayInterval(
	SimpleInterval(500yards, 100%, warmup)
)
IF (Avg): 1
TSS: 11.8
TSS2: 11.8
Dominant Unit: %
Pretty Print:
	* 500 warmup on 7'05''
Duration Unit: yards
Yards: 500
Duration (Sec): 425
----------------------------
Input: 
(500yards, 90, 50s, warmup)
Normalized: 
(500yards, 90, warmup, 50sec)
AST: 
ArrayInterval(
	SimpleInterval(500yards, 90%, warmup, 50sec)
)
IF (Avg): 0.9000000000000001
TSS: 10.6
TSS2: 11.8
Output Unit: /100yards
Pretty Print:
	* 500 warmup on 7'52'' off 8'42'' (1:34/100yards)
Duration Unit: yards
Yards: 500
Duration (Sec): 522.2222222222222
----------------------------
Input: 
(100yards, 100, 10s, strong)
Normalized: 
(100yards, 100, strong, 10sec)
AST: 
ArrayInterval(
	SimpleInterval(100yards, 100%, strong, 10sec)
)
IF (Avg): 1
TSS: 2.4
TSS2: 2.6
Output Unit: /100yards
Pretty Print:
	* 100 strong on 1'25'' off 1'35''
Duration Unit: yards
Yards: 100
Duration (Sec): 95
----------------------------
Input: 
(500yards, +10, 50s, warmup)
Normalized: 
(500yards, +10s, warmup, 50sec)
AST: 
ArrayInterval(
	SimpleInterval(500yards, CSS+10, warmup, 50sec)
)
IF (Avg): 0.8947368421052629
TSS: 10.6
TSS2: 11.7
Output Unit: /100yards
Pretty Print:
	* 500 warmup on 7'55'' off 8'45'' (1:35/100yards)
Duration Unit: yards
Yards: 500
Duration (Sec): 525.0000000000001
----------------------------
Input: 
(500yards, 50s, warmup)
Normalized: 
(500yards, 88, warmup, 50sec)
AST: 
ArrayInterval(
	SimpleInterval(500yards, 88%, warmup, 50sec)
)
IF (Avg): 0.88
TSS: 10.4
TSS2: 11.7
Output Unit: /100yards
Pretty Print:
	* 500 warmup @ 1:37/100yards w/ 50'' rest
Duration Unit: yards
Yards: 500
Duration (Sec): 532.9545454545454
----------------------------
Input: 
(500yards, -5, 50s, warmup)
Normalized: 
(500yards, +-5s, warmup, 50sec)
AST: 
ArrayInterval(
	SimpleInterval(500yards, CSS-5, warmup, 50sec)
)
IF (Avg): 1.0624999999999998
TSS: 12.5
TSS2: 14.6
Output Unit: /100yards
Pretty Print:
	* 500 warmup on 6'40'' off 7'30'' (1:20/100yards)
Duration Unit: yards
Yards: 500
Duration (Sec): 450
----------------------------
Input: 
(500yards, -10, 50s, warmup)
Normalized: 
(500yards, +-10s, warmup, 50sec)
AST: 
ArrayInterval(
	SimpleInterval(500yards, CSS-10, warmup, 50sec)
)
IF (Avg): 1.1333333333333333
TSS: 13.4
TSS2: 14.7
Output Unit: /100yards
Pretty Print:
	* 500 warmup on 6'14'' off 7'04'' (1:15/100yards)
Duration Unit: yards
Yards: 500
Duration (Sec): 424.99999999999994
----------------------------
Input: 
(500yards, 50s, warmup)
Normalized: 
(500yards, 88, warmup, 50sec)
AST: 
ArrayInterval(
	SimpleInterval(500yards, 88%, warmup, 50sec)
)
IF (Avg): 0.88
TSS: 10.4
TSS2: 11.7
Dominant Unit: %
Pretty Print:
	* 500 warmup @ 88% w/ 50'' rest
Duration Unit: yards
Yards: 500
Duration (Sec): 532.9545454545454
----------------------------
Input: 
(500yards, +10, 50s, warmup)
Normalized: 
(500yards, +10s, warmup, 50sec)
AST: 
ArrayInterval(
	SimpleInterval(500yards, CSS+10, warmup, 50sec)
)
IF (Avg): 0.8947368421052629
TSS: 10.6
TSS2: 11.7
Dominant Unit: offset-seconds
Pretty Print:
	* 500 warmup on 7'55'' off 8'45'' (+10)
Duration Unit: yards
Yards: 500
Duration (Sec): 525.0000000000001
----------------------------
Input: 
(100yards, -5, 15s, strong)
Normalized: 
(100yards, +-5s, strong, 15sec)
AST: 
ArrayInterval(
	SimpleInterval(100yards, CSS-5, strong, 15sec)
)
IF (Avg): 1.0624999999999998
TSS: 2.5
TSS2: 3.1
Dominant Unit: offset-seconds
Pretty Print:
	* 100 strong on 1'20'' off 1'35''
Duration Unit: yards
Yards: 100
Duration (Sec): 95.00000000000001
----------------------------
Input: 
(100yards, +0, 20s, strong)
Normalized: 
(100yards, +0s, strong, 20sec)
AST: 
ArrayInterval(
	SimpleInterval(100yards, CSS, strong, 20sec)
)
IF (Avg): 1
TSS: 2.4
TSS2: 2.9
Dominant Unit: offset-seconds
Pretty Print:
	* 100 strong on 1'25'' off 1'45''
Duration Unit: yards
Yards: 100
Duration (Sec): 105
----------------------------
Input: 
(100yards, -0, 5s, strong)
Normalized: 
(100yards, +0s, strong, 5sec)
AST: 
ArrayInterval(
	SimpleInterval(100yards, CSS, strong, 5sec)
)
IF (Avg): 1
TSS: 2.4
TSS2: 2.5
Dominant Unit: offset-seconds
Pretty Print:
	* 100 strong on 1'25'' off 1'30''
Duration Unit: yards
Yards: 100
Duration (Sec): 90
----------------------------
Input: 
(400yards, free)
Normalized: 
(400yards, 88, free)
AST: 
ArrayInterval(
	SimpleInterval(400yards, 88%, free)
)
IF (Avg): 0.88
TSS: 8.3
TSS2: 8.5
Dominant Unit: %
Pretty Print:
	* 400 free @ 88%
Duration Unit: yards
Yards: 400
Duration (Sec): 386.3636363636364
----------------------------
Input: 
(300yards, pull)
Normalized: 
(300yards, 88, pull)
AST: 
ArrayInterval(
	SimpleInterval(300yards, 88%, pull)
)
IF (Avg): 0.88
TSS: 6.2
TSS2: 6.4
Dominant Unit: %
Pretty Print:
	* 300 pull @ 88%
Duration Unit: yards
Yards: 300
Duration (Sec): 289.77272727272725
----------------------------
Input: 
(300yards, pull, 10s)
Normalized: 
(300yards, 88, pull, 10sec)
AST: 
ArrayInterval(
	SimpleInterval(300yards, 88%, pull, 10sec)
)
IF (Avg): 0.88
TSS: 6.2
TSS2: 6.6
Dominant Unit: %
Pretty Print:
	* 300 pull @ 88% w/ 10'' rest
Duration Unit: yards
Yards: 300
Duration (Sec): 299.77272727272725
----------------------------
Input: 
(300yards, +0, Tempo trainer)
Normalized: 
(300yards, +0s, Tempo trainer)
AST: 
ArrayInterval(
	SimpleInterval(300yards, CSS, Tempo trainer)
)
IF (Avg): 1
TSS: 7.1
TSS2: 7.1
Output Unit: /25yards
Pretty Print:
	* 300 Tempo trainer on 4'15'' (0:21/25yards)
Duration Unit: yards
Yards: 300
Duration (Sec): 255
----------------------------
Input: 
(5min, 40)
(30sec, 40, easy)
8[(30s, descend 1-4), (10s, 40)]
(30sec, 40, put door on 3)
10[(60s, 100, LC), (20s, easy)]
(30sec, 40, put door on 1)
10[(120s, 85), (20sec, 40, easy)]
(2min, 40, easy)
5[(30s, 110, fast), (30s, easy)]
(5min, 40, easy)
-Normalized: 
(5min, 40)
(30sec, 40, easy)
8[(30sec, 88, descend 1-4), (10sec, 40)]
(30sec, 40, put door on 3)
10[(1min, 100, LC), (20sec, 88, easy)]
(30sec, 40, put door on 1)
10[(2min, 85), (20sec, 40, easy)]
(2min, 40, easy)
5[(30sec, 110, fast), (30sec, 88, easy)]
(5min, 40, easy)
AST: 
ArrayInterval(
	SimpleInterval(5min, 40%, )
	SimpleInterval(30sec, 40%, easy)
	RepeatInterval(count=8, 
		SimpleInterval(30sec, 88%, descend 1-4)
		SimpleInterval(10sec, 40%, )
	)
	SimpleInterval(30sec, 40%, put door on 3)
	RepeatInterval(count=10, 
		SimpleInterval(1min, 100%, LC)
		SimpleInterval(20sec, 88%, easy)
	)
	SimpleInterval(30sec, 40%, put door on 1)
	RepeatInterval(count=10, 
		SimpleInterval(2min, 85%, )
		SimpleInterval(20sec, 40%, easy)
	)
	SimpleInterval(2min, 40%, easy)
	RepeatInterval(count=5, 
		SimpleInterval(30sec, 110%, fast)
		SimpleInterval(30sec, 88%, easy)
	)
	SimpleInterval(5min, 40%, easy)
)
IF (Avg): 0.7924916246917765
TSS: 63.3
TSS2: 63.9
Dominant Unit: %
Pretty Print:
	* 5' easy
	* 30'' easy @ 40%
	* 8 x 30'' descend 1-4 on 30'' - 10'' easy
	* 30'' put door on 3 @ 40%
	* 10 x 1' LC on 1' - 20'' easy
	* 30'' put door on 1 @ 40%
	* 10 x 2' on 2' - 20'' easy
	* 2' easy @ 40%
	* 5 x 30'' fast on 30'' - 30'' easy
	* 5' easy @ 40%
Duration (Sec): 3630
----------------------------
